### PR TITLE
Feature/5 Авторизация: backend

### PR DESCRIPTION
**Pull request by okostyuk**
#5 Авторизация: backend

### CODE REVIEW

File: server/Entities/Migrations/20200507183659_Initial.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                {
+                    table.PrimaryKey("PK_users", x => x.Id);
+                });
+
+            migrationBuilder.InsertData(
+                table: "users",
+                columns: new[] { "Id", "Email", "Password" },
+                values: new object[,]
+                {
+                    { -1, "test1@users.com", "d53tbpepSAOL9x9Xo2ea/LLhaJSgmyO0CCXFrYcdk66s3icf5o3lPGejbtagCjdD" },
```
Comment (demosglok) [inline server/Entities/Migrations/20200507183659_Initial.cs:28]: wont negative ID cause any problems later on? even if they are "test"

File: server/Entities/RepositoryContext.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+        protected override void OnConfiguring(DbContextOptionsBuilder options)
+            => options.UseSqlServer(
+                "Server=192.168.1.128,51528\\SQLEXPRESS;Database=master;User Id=sa;Password=server;"
+                );
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<User>().HasData(
+                new User { Id = -1, Email = "test1@users.com", Password = CryptoUtils.HashPassword("test1") },
```
Comment (demosglok) [inline server/Entities/RepositoryContext.cs:23]: you have this in migration, don't you?

File: server/WebAPI/Startup.cs [hunk 3]
```diff
@@ -39,6 +45,25 @@ public void ConfigureServices(IServiceCollection services)
                                   });
             });
 
+            services.AddDbContext<RepositoryContext>(options => 
+                options.UseSqlServer(Configuration.GetConnectionString("mssqlConnection")));
+            services.AddSingleton(Configuration);
+            services.AddScoped<UserRepository>();
+            services.AddScoped<UserService>();
+            services.AddSingleton<AuthService>();
```
Comment (demosglok) [inline server/WebAPI/Startup.cs:53]: just curoius, why do we want auth service as sigleton while others are scoped?

**Review by demosglok — COMMENTED** [review]
looks good to me, fine to merge

**Review by demosglok — APPROVED** [review]
lgtm

File: server/Entities/Migrations/20200507183659_Initial.cs (hunk 1 above)
Comment (okostyuk) [reply server/Entities/Migrations/20200507183659_Initial.cs:28]: I don`t think so. INT data types in mssql and c# supports negative values, and at least it was recomended by EF console tool - to use negative ids for seeded items

File: server/Entities/RepositoryContext.cs (hunk 2 above)
Comment (okostyuk) [reply server/Entities/RepositoryContext.cs:23]: Not sure do I understand the question correctly, but generated Migration class contains Instert statement for all seeded data:
```
migrationBuilder.InsertData(
                table: "users",
                columns: new[] { "Id", "Email", "Password" },
                values: new object[,]
                {
                    { -1, "test1@users.com", "d53tbpepSAOL9x9Xo2ea/LLhaJSgmyO0CCXFrYcdk66s3icf5o3lPGejbtagCjdD" }
```

File: server/WebAPI/Startup.cs (hunk 3 above)
Comment (okostyuk) [reply server/WebAPI/Startup.cs:53]: That`s because now AuthService doesn`t use any Scoped instances unlike UserService

**Pull request merged**